checker/src/main/java/org/checkerframework/checker/nullness/KeyForAnnotatedTypeFactory.java

```diff
@@ -218,19 +234,71 @@ public boolean isSubtype(AnnotationMirror subAnno, AnnotationMirror superAnno) {
+     * one formal parameter of Object type.
+     *
+     * @param n the method invocation
+     * @param methodName the method to check for, which has one formal parameter
+     */
+    protected boolean isInvocationOfMapMethodWithOneObjectParameter(
+            MethodInvocationNode n, String methodName) {
+        return isInvocationOfMapMethod(n, methodName, PARAMS_OBJECT);
+    }
+
+    /**
+     * Returns true if the node is an invocation of the named method in the Map interface, which has
+     * two formal parameters of Object type.
+     *
+     * @param n the method invocation
+     * @param methodName the method to check for, which has two formal parameters
+     */
+    protected boolean isInvocationOfMapMethodWithTwoObjectParameters(
+            MethodInvocationNode n, String methodName) {
+        return isInvocationOfMapMethod(n, methodName, PARAMS_OBJECT_OBJECT);
+    }
+
+    /**
+     * Returns true if the node is an invocation of the named method in the Map interface.
+     *
+     * @param n the method invocation
+     * @param methodName the method to check for
+     * @param paramTypes the parameter types in methodName's declaration, to resolve overloading
+     */
+    protected boolean isInvocationOfMapMethod(
+            MethodInvocationNode n, String methodName, List<TypeMirror> paramTypes) {
+        String invokedMethodName = getMethodName(n);
+
         // First verify if the method name is correct. This is an inexpensive check.
-        if (invokedMethod.equals(methodName)) {
-            // Now verify that the receiver of the method invocation is of a type
-            // that extends that java.util.Map interface. This is a more expensive check.
-            TypeMirror receiverType = types.erasure(n.getTarget().getReceiver().getType());
+        if (!invokedMethodName.equals(methodName)) {
+            return false;
+        }
 
-            if (types.isSubtype(receiverType, erasedMapType)) {
-                return true;
+        // Verify the argument types.  These are more expensive checks.
+
+        // The receiver's type must extend the java.util.Map interface.
+        TypeMirror receiverType = types.erasure(n.getTarget().getReceiver().getType());
+        if (!types.isSubtype(receiverType, erasedMapType)) {
+            return false;
+        }
+
+        // Also verify the other argument types
+        List<Node> args = n.getArguments();
+        if (args.size() != paramTypes.size()) {
+            return false;
+        }
+        for (int i = 0; i < args.size(); i++) {
+            Node arg = args.get(i);
+            TypeMirror erasedArgType = types.erasure(n.getArguments().get(i).getType());
+            TypeMirror paramType = paramTypes.get(i);
+            // This test isn't right, because there could be a more specific overload that is
+            // applied.  Exact equality wouldn't be right either.
```

Comment:
This comment is a bit confusing.
This test isn't right, but it's right enough? Or are there things that need improvement?
Can you clarify why exact equality isn't right?
At the moment the passed in parameter types are always `j.l.Object`, so the tests are trivially false. Would just passing the number of arguments be enough to disambiguate overloading?